packages/blocks/src/api/raw-handling/phrasing-content-reducer.js

```diff
@@ -22,7 +28,10 @@ export default function( node, doc ) {
 			wrap( doc.createElement( 'em' ), node );
 		}
 
-		if ( textDecorationLine === 'line-through' ) {
+		// Some DOM implementations (Safari, JSDom) don't support
+		// style.textDecorationLine, so we check style.textDecoration as a
+		// fallback.
+		if ( textDecorationLine === 'line-through' || includes( textDecoration, 'line-through' ) ) {
```

Comment:
I get where you're coming from but I don't really want to stray into the territory of implementing a CSS parser here. This simple check is correct because, like you say, the only scenario in which `text-decoration` would contain the substring `'line-through'` is iff `text-decoration-line` is `'line-through'`.